worlds/sc2/PoolFilter.py

```diff
@@ -69,21 +69,39 @@ def filter_missions(world: World) -> Dict[MissionPools, List[SC2Mission]]:
         return mission_pools
 
     # Finding the goal map
-    goal_priorities = {campaign: get_campaign_goal_priority(campaign, excluded_missions) for campaign in enabled_campaigns}
-    goal_level = max(goal_priorities.values())
-    candidate_campaigns: List[SC2Campaign] = [campaign for campaign, goal_priority in goal_priorities.items() if goal_priority == goal_level]
-    candidate_campaigns.sort(key=lambda it: it.id)
-    goal_campaign = world.random.choice(candidate_campaigns)
-    primary_goal = campaign_final_mission_locations[goal_campaign]
-    if primary_goal is None or primary_goal.mission in excluded_missions:
-        # No primary goal or its mission is excluded
-        candidate_missions = list(campaign_alt_final_mission_locations[goal_campaign].keys())
-        candidate_missions = [mission for mission in candidate_missions if mission not in excluded_missions]
-        if len(candidate_missions) == 0:
-            raise Exception("There are no valid goal missions. Please exclude fewer missions.")
-        goal_mission = world.random.choice(candidate_missions)
+    goal_mission: Optional[SC2Mission] = None
+    if mission_order_type in campaign_depending_orders:
```

Comment:
Does this section of code only determine the goal mission, and not determine which mission slot is the final one?  If so, couldn't you add `and get_option_value('shuffle_campaigns') is False` to this check and get Figment's desired behavior (hardest mission available when campaigns are shuffled)?

Reply:
this determines the goal's mission and if the order is camapign depending, then there are multiple final slots and then the one that belongs to the final mission's campaign is picked

Reply:
See in `Regions.py`:
```python
    if mission_order_type in campaign_depending_orders:
        # Do slot removal per campaign
        for campaign in enabled_campaigns:
            campaign_mission_pool = [mission for mission in mission_pool if mission.campaign == campaign]
            campaign_mission_pool_size = len(campaign_mission_pool)

            removals = len(mission_order[campaign]) - campaign_mission_pool_size

            for mission in mission_order[campaign]:
                # Removing extra missions if mission pool is too small
                if 0 < mission.removal_priority <= removals:
                    mission_slots.append(SC2MissionSlot(campaign, None))
                elif mission.type == MissionPools.FINAL:
                    if campaign == final_mission.campaign:
                        # Campaign is elected to be goal
                        mission_slots.append(SC2MissionSlot(campaign, final_mission))
                    else:
                        # Not the goal, find the most difficult mission in the pool and set the difficulty
                        campaign_difficulty = max(mission.pool for mission in campaign_mission_pool)
                        mission_slots.append(SC2MissionSlot(campaign, campaign_difficulty))
                else:
                    mission_slots.append(SC2MissionSlot(campaign, mission.type))
```

Reply:
if all very hards are excluded, it could place the goal into Prophecy lane if In Utter Darkness would be picked

Reply:
I'd leave changing that mechanism to mission plando at it'd need to change more things

Reply:
Fair enough.